packages/ra-ui-materialui/src/list/filter/FilterButton.spec.tsx

```diff
@@ -124,5 +124,27 @@ describe('<FilterButton />', () => {
                 ).toBeNull();
             });
         });
+
+        it('should not display save query in filter button', () => {
+            const { queryByText } = render(
+                <AdminContext theme={theme}>
+                    <FilterButton
+                        {...defaultProps}
+                        filterValues={{ title: 'foo' }}
+                        filters={[
+                            <TextInput source="Returned" label="Returned" />,
+                        ]}
+                        disableSaveQuery
+                    />
+                </AdminContext>
+            );
+            expect(
+                screen.queryByLabelText('ra.action.add_filter')
+            ).not.toBeNull();
+
+            fireEvent.click(screen.getByLabelText('ra.action.add_filter'));
+
+            expect(queryByText('ra.saved_queries.new_label')).toBeNull();
```

Comment:
This test will pass even if the feature isn't implemented, because there is a delay between the filter button click and the filter menu to show. You should wait for the menu to appear (with `await findByText('Returned')`) before checking that the Save Query menu item isn't there.

Reply:
From the tests I fixed I didn't notice any problems and leave it without, but it makes a lot of sense that. Nice Catch


Done https://github.com/marmelab/react-admin/pull/8113/commits/70ee27540b4a005c93c1bde411d05bc954ae981e